src/v/redpanda/admin_server.cc

```diff
@@ -537,4 +583,86 @@ void admin_server::register_partition_routes() {
                   std::move(p));
             });
       });
+
+    // make sure to call reset() before each use
```

Comment:
Would it make sense to release the resources after they're done with / scope exit?

Reply:
@BenPope  what did you have in mind? my mental model here is that the validator needs to be alive until no more admin http requests can be handled which would be on system shutdown

Reply:
Understood, I was suggesting that reset is called on scope exit, or otherwise eagerly, so that it doesn't hold onto resources, and there's no attempt to use it in this function after a yield to the scheduler.

Reply:
Perhaps:
```cpp
          if (!doc.Accept(set_replicas_validator.validator)) {
              set_replicas_validator.validator.Reset();
              throw ss::httpd::bad_request_exception(
                "Replica set json is invalid");
          }
          set_replicas_validator.validator.Reset();
 ```

Reply:
Ahh, I see. I just assumed it didn't hold onto resources. I'll also add this.

Reply:
https://github.com/vectorizedio/redpanda/issues/1408